helm-app-operator/pkg/helm/controller/reconcile.go

```diff
@@ -0,0 +1,70 @@
+package controller
+
+import (
+	"context"
+	"log"
+
+	"github.com/operator-framework/helm-app-operator-kit/helm-app-operator/pkg/helm/installer"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type helmOperatorReconciler struct {
+	GVK       schema.GroupVersionKind
+	Installer installer.Installer
+	Client    client.Client
+}
+
+// assert interface
+var _ reconcile.Reconciler = &helmOperatorReconciler{}
+
+// Reconcile handles events by installing, updating, or uninstalling the
+// associated helm releases.
+func (r *helmOperatorReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	u := &unstructured.Unstructured{}
+	u.SetGroupVersionKind(r.GVK)
+	u.SetNamespace(request.Namespace)
+	u.SetName(request.Name)
+
+	err := r.Client.Get(context.TODO(), request.NamespacedName, u)
+	if err != nil {
+		// If the CR is not found, it must have just been deleted. Uninstall it.
+		if apierrors.IsNotFound(err) {
+			u, err = r.Installer.UninstallRelease(u)
+			if err != nil {
+				log.Print(err)
```

Comment:
It probably makes the most sense to go with the outcome of operator-framework/operator-sdk#503, since there's discussion and voting there about the preferences and technical merits of various loggers (and just for the sake of consistency). I'll open a new issue to follow up once that decision is made.

Reply:
New issue #44 to track logging updates.